all-of-us/workbench
# [risk=low][RW-9065] Allow machine type configuration during User App creation

Tested locally:
* create a SAS app with a nonstandard configuration
* observe that the cost updates when the configuration does
* observe that changing the machine type is not possible during app creation, running, or deletion
* observe that configuration is not allowed when an app exists of a different type (i.e. can't configure SAS if RStudio exists)
* observe that I can open SAS when I configure it as `n1-highmem-4` (normal is `n1-standard-4`)
* observe that changing the machine type is possible after deletion
* observe that deleting an app with configuration X does not require any waiting before creating an app with configuration Y
* observe that an altered app configuration does not persist after deletion (creating a new app starts with the default settings - note: this differs from Jupyter)

Won't do in this story:
* Determine which configurations are incompatible with specific app types 


When no app exists, the user can configure
<img width="901" alt="can config" src="https://github.com/all-of-us/workbench/assets/2701406/c5222c02-6c5a-44e2-a357-1cbe28cea49d">

When the app is running, the user cannot configure
<img width="927" alt="same-app-running" src="https://github.com/all-of-us/workbench/assets/2701406/294b3132-2cc7-4879-bc67-fd62dc2fab28">

When the app is deleting, the user cannot configure
<img width="959" alt="app-deleting" src="https://github.com/all-of-us/workbench/assets/2701406/85e60e65-b6ae-4f22-872e-cc6d0d1bf187">

When **a different app type** is running, the user cannot configure.  The purpose of this is to prevent multiple nodepools.
<img width="959" alt="other-app-running" src="https://github.com/all-of-us/workbench/assets/2701406/e7a98507-b181-42a0-9b7f-2832eae99893">


---
**PR checklist**

- One or more of the following is true:
